lib/use_case/relationship/destroy.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require "use_case/base"
+require "errors"
+
+module UseCase
+  module Relationship
+    class Destroy < UseCase::Base
+      option :current_user, type: Types::Coercible::String | Types.Instance(User)
+      option :target_user, type: Types::Coercible::String | Types.Instance(User)
+      option :type, type: Types::RelationshipTypes
+
+      def call
+        source_user = find_source_user
+        target_user = find_target_user
+
+        source_user.public_send("un#{type}", target_user)
+
+        {
+          status:   204,
+          resource: nil,
+          extra:    {
+            target_user: target_user
+          }
+        }
+      end
+
+      private
+
+      def find_source_user
+        ::User.find_by!(screen_name: current_user)
```

Comment:
...interestingly enough, in this UseCase `current_user` could be a `::User` object :D 

so you could just:
```suggestion
      def find_source_user
        return current_user if current_user.is_a?(::User)

        ::User.find_by!(screen_name: current_user)
```